### PR TITLE
Fix global VmIdHandles initialization fiasco and build failure 

### PR DESCRIPTION
**Pull request by mathetake**
first two commits are cherry picked from #114

### CODE REVIEW

File: src/vm_id_handle.cc
Comment (mathetake) [inline src/vm_id_handle.cc:28]: suspicious about this mutex for Windows test failure

Comment (PiotrSikora) [reply src/vm_id_handle.cc:28]: Yeah, I think mutex fails under the same fiasco.

Comment (mathetake) [reply src/vm_id_handle.cc]: so should we fix https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/wasm.cc#L44 as well? though these are never access in the initialization phase since they are not accessed until createWasm is called

**Comment by mathetake** [comment]
TODO: after passing CI in Envoy, will use https://github.com/envoyproxy/envoy/blob/master/source/common/common/macros.h#L38-L42 this macro for clarity

**Comment by mathetake** [comment]
@lizan would like you to review this since I followed your suggestion about global variables

**Comment by mathetake** [comment]
the mutex was the culprit of Windows failure actually.

File: src/vm_id_handle.cc
Comment (PiotrSikora) [reply src/vm_id_handle.cc:26]: No, those are fine. The issue here is `global_shared_data` that requires all the `vm_handle` objects.

File: src/vm_id_handle.cc
Comment (PiotrSikora) [inline src/vm_id_handle.cc:24]: Please re-add the empty line.

File: src/vm_id_handle.cc
Comment (PiotrSikora) [inline src/vm_id_handle.cc:47]: Debug leftover.

File: src/shared_data.cc
Comment (PiotrSikora) [inline src/shared_data.cc:34]: Why do we need this mutex here? Also, it gets out of scope, so access to the returned object won't be guarded.

File: src/vm_id_handle.cc
Comment (lizan) [inline src/vm_id_handle.cc:26]: you should do the same pattern for the mutex as well.

File: src/vm_id_handle.cc
Comment (lizan) [inline src/vm_id_handle.cc:28]: return reference? because they won't be nullptr anyway.

**Review by lizan — CHANGES_REQUESTED** [review]

File: src/shared_data.cc
Comment (mathetake) [reply src/shared_data.cc:34]: I just misunderstand that static local initialization is not thread safe. will fix this

**Review by PiotrSikora — APPROVED** [review]
Thanks, this looks good. I'll wait for @lizan's approval before merging.

**Review by lizan — APPROVED** [review]

**Pull request merged**
